### PR TITLE
My booking page

### PR DESCRIPTION
**Pull request by adambrodin**
- Implemented API requests to fetch booking data & cancel booking
- Error message when booking details are invalid
- Minor styling
- Refactored code to use DTO as data model
- Minor refactoring

### CODE REVIEW

**Comment by mzarabi** [comment]
Ser bra ut men försökte testa de på deployen då hände inget när jag tryckte på Sök, borde väl få nåt fel då jag skrev randomuppgifter?

File: client/src/MyBookingsPage/SearchBookingComponent.tsx [hunk 1]
```diff
@@ -1,58 +1,45 @@
...
+
   function search() {
-    validateBookingId(bookingId);
-    validateEmail(email);
-    setSearchReady(true);
+    let idPassedValidation = validateBookingId(bookingId);
+    let emailPassedValidation = validateEmail(email);
+    setBookingIdError(!idPassedValidation);
+    setBookingIdErrorMessage(
+      idPassedValidation ? "" : "Ogiltigt referensnummer. "
```
Comment (Spuute) [inline client/src/MyBookingsPage/SearchBookingComponent.tsx]: I think we should use 'Bokningsnummer' instead of 'referensnummer'

**Comment by Spuute** [comment]
I think this looks good, but as I mentioned in a comment, i personally think we should use 'bokningsnummer' instead of 'referensnummer', or if you prefer 'referensnummer' I will change the email function from 'bokningsnummer' to 'referensnummer'

File: client/src/MyBookingsPage/SearchBookingComponent.tsx (hunk 1 above)
Comment (adambrodin) [reply client/src/MyBookingsPage/SearchBookingComponent.tsx]: Name has been changed to 'bokningsnummer'.

File: client/src/MyBookingsPage/SearchBookingComponent.tsx [hunk 2]
```diff
@@ -1,66 +1,53 @@
...
-    validateEmail(email);
-    setSearchReady(true);
+    let idPassedValidation = validateBookingId(bookingId);
+    let emailPassedValidation = validateEmail(email);
+    setBookingIdError(!idPassedValidation);
+    setBookingIdErrorMessage(
+      idPassedValidation ? "" : "Ogiltigt bokningsnummer. "
+    );
+    setEmailError(!emailPassedValidation);
+    setEmailErrorMessage(emailPassedValidation ? "" : "Ogiltigt email. ");
```
Comment (Spuute) [inline client/src/MyBookingsPage/SearchBookingComponent.tsx]: To be really picky 'Ogiltlig email.' without the ending t :)

Comment (adambrodin) [reply client/src/MyBookingsPage/SearchBookingComponent.tsx]: Grammar has been fixed.

**Review by Spuute — APPROVED** [review]
Nice work!

**Pull request merged**